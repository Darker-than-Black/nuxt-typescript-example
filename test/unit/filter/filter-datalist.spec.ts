import vClickOutside from 'v-click-outside'
import VueTestInstance from '~/test/factory/vue-instance'
import FilterDatalist from '~/components/filter/filter-datalist.vue'

describe('filter-select.vue', () => {
  let wrapper: any
  const options: string[] = ['asd', 'lorem', 'ipsum']

  beforeEach(() => {
    wrapper = VueTestInstance(FilterDatalist, {
      isShallow: false,
      modules: [vClickOutside],
      options: {
        propsData: {
          showOptions: true,
          label: 'Filter Select',
          listLink: 'linkToDataLink',
          value: '',
          options
        }
      }
    })
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check input function', () => {
    expect(wrapper.vm.input(options[1])).toBe(options[1])
  })
})
