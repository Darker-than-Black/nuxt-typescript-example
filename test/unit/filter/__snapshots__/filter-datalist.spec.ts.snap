// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter-select.vue should match the snapshot 1`] = `
<div class="d-inline-block" showoptions="true">
  <div><label for="Filter Select" class="title__field">
      Filter Select
      <!----></label>
    <div class="position-relative"><input id="Filter Select" name="Filter Select" placeholder="Оберіть зі списку" type="text" list="linkToDataLink" class="input input__padding-right"> <img src="/icons/Arrow-down.svg" alt="arrow" class="img filter__icon"></div>
  </div> <datalist id="linkToDataLink">
    <option value="asd"></option>
    <option value="lorem"></option>
    <option value="ipsum"></option>
  </datalist>
</div>
`;
